blog/2023-11-angular17/README.md

```diff
@@ -0,0 +1,427 @@
+
+
+### Was passiert mit den Direktiven?
+
+Die bisherigen Direktiven bleiben zunächst erhalten und können parallel zum Control Flow verwendet werden.
+Sie müssen Ihre Anwendungen also nicht sofort migrieren, sondern können auch weiterhin den gewohnten Ablauf mit den Direktiven nutzen.
+Ebenso ist ein Mischbetrieb möglich, sodass z. B. neue Features mit dem neuen Control Flow ausgestattet werden können.
+
+Grundsätzlich empfehlen wir Ihnen jedoch, in den nächsten Monaten schrittweise zur neuen Control-Flow-Syntax zu migrieren.
+Es ist möglich, dass die Direktiven `NgIf`, `NgFor` und `NgSwitch` in einer zukünftigen Major-Version als *deprecated* markiert werden.
+<!-- In einigen Jahren werden diese Direktiven wahrscheinlich aus dem Framework entfernt. -->
+
+Angular stellt übrigens ein Skript bereit, um die Templates auf den neuen Control Flow zu migrieren:
+
+```bash
+ng generate @angular/core:control-flow
+```
+
+Bitte beachten Sie, dass die Migration nicht vollständig automatisch funktioniert.
+Die Ergebnisse müssen stets manuell geprüft und nachgebessert werden.
+In unseren ersten Experimenten war die Automigration hilfreich, hat aber nicht alle Fälle korrekt erfasst.
+
+
+## Deferrable Views mit `@defer`
+
+<!-- siehe https://github.com/angular/angular/discussions/50716 -->
+
+Mit dem neuen Control Flow wird ein sehr nützliches neues Feature eingeführt: der `@defer`-Block.
+
+Die Hauptaufgabe des `@defer`-Blocks besteht darin, Inhalte verzögert zu laden. Egal ob es sich um eine Komponente, eine Direktive oder eine Pipe handelt, wenn sie in einem solchen Block platziert werden, lädt Angular sie nur unter bestimmten Bedingungen oder bei bestimmten Ereignissen. Das ist besonders nützlich, um die Leistung zu optimieren, insbesondere wenn bestimmte Komponenten nicht sofort benötigt werden oder für die anwendende Person noch nicht sichtbar sind.
+
+```html
+<p>Dieser foglende Inhalt wird später geladen!</p>
+
+@defer {
+  <book-details [book]="myBook" />
+}
+@loading {
+  <span>Lade Abhängigkeiten …</span>
+} @placeholder {
+  <span>Inhalt Inhalt wird gerendert …</span>
+} @error {
+  <span>Es kam zu einem Fehler!</span>
+}
+```
+
+Folgende Helfer stehen zur Verfügung
+
+* `@loading`: Zeigt den angegebenen Inhalt während des Ladens von Abhängigkeiten an.
+* `@placeholder`: Zeigt den bereitgestellten Inhalt als vorübergehende Anzeige, bis der eigentliche Inhalt vollständig gerendert ist. Der Platzhalter muss immer angegeben werden! 
+* `@error`: Zeigt den angegebenen Inhalt an, falls ein Problem beim Laden der Abhängigkeiten des Inhalts auftritt.
+
+
+Uns stehen jetzt eine Sammlung von Triggern zu Verfügung, die steuern, wann Angular den Inhalt laden und rendern soll:
+
+### Loading Trigger: `on viewport`
+
+Der Inhalt soll nachladen, wenn das Element sichtbar wird:
+
+```html
+@defer (on viewport) {
+  <p>Dieser Inhalt wird später geladen!</p>
+}
+@placeholder {
+  <p>Inhalt wird gerendert …</p>
```

Comment:
Wo liest du das in dem RFC? "Rendering" heißt für mich der Zeitraum zwischen erfolgreichem Laden und der fertigen Darstellung. Praktisch wird der Placeholder die ganze Zeit angezeigt, bevor der eigentliche Inhalt zu sehen ist – bei einem Timer von 3s also nach 3s und ein bisschen.

Reply:
> The @placeholder is an optional block that declares content to show before the defer block is triggered.

Hat nichts nicht dem eigentlich Rendering zu tun, sondern es ist der Platzhalter für das eigentliche Element, bevor das da ist (durch welchen Trigger auch immer)

Reply:
`loading` verstehe ich so: Es wird angezeigt, während die View tatsächlich geladen und gerendert wird.

Reply:
```
@defer (on interaction) {
  <p>WIRD DEFERRED GELADEN</p>
}
@loading (minimum 1s) { <p>WIRD GELADEN</p> }
@placeholder { <p>IST NOCH NICHT GELADEN</p> }
```

Reply:
https://github.com/angular-buch/website-articles/assets/1683147/ae1ce2f0-d9b9-4a97-b1fb-5134a95f6087

Reply:
Ich hab das im Text überarbeitet

Reply:
Kann gut sein, das ich das falsch verstanden habe, hab den Code nicht getestet